doc/high-level.md

```diff
@@ -0,0 +1,83 @@
+
+| Element Type             | Class Suffix           |
+|--------------------------|------------------------|
+| `SINGLE-FLOAT`           | `SINGLE-FLOAT`         |
+| `DOUBLE-FLOAT`           | `DOUBLE-FLOAT`         |
+| `(COMPLEX SINGLE-FLOAT)` | `COMPLEX-SINGLE-FLOAT` |
+| `(COMPLEX DOUBLE-FLOAT)` | `COMPLEX-DOUBLE-FLOAT` |
+| `(SIGNED-BYTE 32)`       | `INT32`                |
+
+The type of the elements of the tensor can be specified with the `:type` keyword, or the constructor will attempt to find an appropriate type from the given arguments. The default element type for a tensor is `DOUBLE-FLOAT`.
+
+The order of the tensor (column-major or row-major) can be specified with the `:order` keyword. This affects the underlying storage of the tensor and will affect how it carries out operations with LAPACK. 
+
+
+## Other Library Equivalents
+
+This table was adapted largely from the [NumPy Equivalent Table](https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html#linear-algebra-equivalents).
+
+### Basic Accessors
+
+| MAGICL         | MATLAB     | NumPy                   | Description                                                   |
+|----------------|------------|-------------------------|---------------------------------------------------------------|
+| `(order a)`    | `ndims(a)` | `ndim(a)` or `a.ndim`   | Get the number of dimensions of the array.                    |
+| `(size a)`     | `numel(a)` | `size(a)` or `a.size`   | Get the number of elements of the array.                      |
+| `(shape a)`    | `size(a)`  | `shape(a)` or `a.shape` | Get the shape of the array.                                   |
+| `(tref a 1 4)` | `a(2,5)`   | `a[1 4]`                | Get the element in the second row, fifth column of the array. |
+
+### Constructors
+
+| MAGICL                                          | MATLAB             | NumPy                             | Description                                                                          |
+|-------------------------------------------------|--------------------|-----------------------------------|--------------------------------------------------------------------------------------|
+| `(from-list '(1d0 2d0 3d0 4d0 5d0 6d0) '(2 3))` | `[ 1 2 3; 4 5 6 ]` | `array([[1.,2.,3.], [4.,5.,6.]])` | Create a 2x3 matrix from given elements.                                             |
+| `(empty '(2 3 4))` or `(const 0d0 '(2 3 4))`    | `zeroes(2,3,4)`    | `zeroes((2,3,4))`                 | Create a 2x3x4 dimensional array of zeroes of double-float element type.             |
+| `(const 1d0 '(3 4))`                            | `ones(3,4)`        | `ones((3,4))`                     | Create a 3x4 dimensional array of ones of double-float element type.                 |
+| `(deye 1d0 '(3 3))`                             | `eye(3)`           | `eye(3)`                          | Create a 3x3 identity array of double-float element type.                            |
```

Comment:
The interface for `deye` is awkward in my opinion. Perhaps you could change the first argument into an optional or keyword argument and also infer the data type from that argument (assuming a double precision 1.0 by default is reasonable and in line with `+default-tensor-type+`). Having another keyword/optional argument for specifying the data type would be good too (à la `dtype` in NumPy).

Additionally, the caller should be able to set which diagonal to fill (via a diagonal index as MATLAB and NumPy do). This is useful when creating tri-diagonal or banded matrices (e.g., when doing finite difference approximations of second order derivatives).

Finally, it would be handy for the `shape` argument to expand to `'(n n)` whenever it receives a fixnum `n` (i.e., default to square matrices in a succinct manner).

Why is it called `deye`? Does "d" stand for double? If so, having a sensible default and inferring or declaring the type using a keyword parameter would strengthen the case for dropping the initial "d".

Example of desirable uses: `(eye 3)`, `(eye '(3 4) :value #C(0.0d0 -1.0d0))`, `(eye 3 :index -1)`, `(eye 4 :type int32`, etc.

Reply:
All features except diagonal banding have been added. Diag bands in #73